src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -194,9 +194,29 @@ export class RelayFeeCalculator {
     return this.queries.getTokenPrice(tokenSymbol);
   }
 
+  /**
+   * Calculate the gas fee as a % of the amount to relay.
+   * @param deposit A valid deposit object to reason about
+   * @param amountToRelay The amount that we should fill the deposit for
+   * @param simulateZeroFill Whether to simulate a zero fill for the gas cost simulation
+   *        A fill of 1 wei which would result in a slow/partial fill.
+   *        You should do this if you're not worried about simulating a proper fill of a deposit
+   *        with a message or if you are worried a fill amount that could exceed the balance of
+   *        the relayer.
+   * @param relayerAddress The relayer that will be used for the gas cost simulation
+   * @param _tokenPrice The token price for normalizing fees
+   * @returns The fee as a % of the amount to relay.
+   * @note Setting simulateZeroFill to true will result on the gas costs being estimated
+   *       on a zero fill. However, the percentage will be returned as a percentage of the
+   *       amount to relay. This is useful for determining the maximum gas fee % that a
+   *       relayer may need to make on a regular fill. You will get differing results if
+   *       a message & recipient contract is provided as this function may not simulate with
+   *       the correct parameters to see a full fill.
+   */
   async gasFeePercent(
     deposit: Deposit,
     amountToRelay: BigNumberish,
+    simulateZeroFill = true,
```

Comment:
why make this default `true`? I would think this should be default `false`

Reply:
Good point -- agreed.

Reply:
My thought is that our default case so far is that we're dealing with non-message fields. However, I am of two minds.

I've updated the PR to make this default to false https://github.com/across-protocol/sdk-v2/pull/445/commits/801cfaac30169e380f6aa5b65191a23cbe423bb4

Reply:
That may be the common case for the API, but other users, like the relayer, would probably want to do this estimate on the un-modified deposit, right? I think the setting that is what an uninformed caller might expect is probably the better default.